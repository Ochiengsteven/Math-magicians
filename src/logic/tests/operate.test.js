import operate from '../operate';

describe('operate function', () => {
  it('adds two numbers', () => {
    expect(operate(2, 3, '+')).toMatchSnapshot();
  });

  it('subtracts two numbers', () => {
    expect(operate(5, 3, '-')).toMatchSnapshot();
  });

  it('multiplies two numbers', () => {
    expect(operate(2, 3, '*')).toMatchSnapshot();
  });

  it('divides two numbers', () => {
    expect(operate(6, 2, '/')).toMatchSnapshot();
  });

  it('handles division by zero', () => {
    expect(operate(6, 0, '/')).toMatchSnapshot();
  });

  it('calculates modulus', () => {
    expect(operate(7, 3, '%')).toMatchSnapshot();
  });

  it('handles modulo by zero', () => {
    expect(operate(7, 0, '%')).toMatchSnapshot();
  });

  it('throws error for unknown operation', () => {
    expect(() => operate(2, 3, '^')).toThrowErrorMatchingSnapshot();
  });
});
