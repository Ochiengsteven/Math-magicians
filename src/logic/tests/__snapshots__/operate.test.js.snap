// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate function adds two numbers 1`] = `"5"`;

exports[`operate function calculates modulus 1`] = `"1"`;

exports[`operate function divides two numbers 1`] = `"3"`;

exports[`operate function handles division by zero 1`] = `"Can't divide by 0."`;

exports[`operate function handles modulo by zero 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`operate function multiplies two numbers 1`] = `"6"`;

exports[`operate function subtracts two numbers 1`] = `"2"`;

exports[`operate function throws error for unknown operation 1`] = `"Unknown operation '^'"`;
